include/tvm/runtime/ndarray.h

```diff
@@ -296,6 +299,19 @@ class NDArray::Container {
     dl_tensor.strides = nullptr;
     dl_tensor.byte_offset = 0;
   }
+
+  Container(void* data,
```

Comment:
can we just use new and set it up in the impl? To minimize the constructor logic in here

Reply:
Why minimize? it creates even more chances to do it wrong code for construction should be constructors

Reply:
There are certain cases where the shape vector is not needed, when converting from DLPack. Anyway if we want to keep it in this way , try to use copy initializers to initialize shape, this will remove the need for move

Reply:
I think we can keep the constructor this way because it's non-trivial to correctly initialize the shape in container and DLTensor in my mind. If there's use case to create Container without shape in the future, we can revisit this or add a new constructor.